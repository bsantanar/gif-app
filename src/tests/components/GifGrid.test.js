import { shallow } from 'enzyme'
import { GifGrid } from '../../components/GifGrid'
import { useFetchGifs } from '../../hooks/useFetchGifs'
jest.mock('../../hooks/useFetchGifs')

describe('tests for <GifGrid />', () => {

    const category = 'Jojo'


    test('should match snapshot', () => {
        useFetchGifs.mockReturnValue({
            data:[],
            loading: false
        })
        const wrapper = shallow(<GifGrid  category={category}/>)
        expect(wrapper).toMatchSnapshot()
    })
    
    test('should show items when img loaded', () => {

        const gifs = [{
            id:'ABC',
            url: 'https://localhost:8000/url',
            title: 'titulo'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })
        const wrapper = shallow(<GifGrid  category={category}/>)

        expect(wrapper.find('p').exists()).toBe(false)
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
    })
    
})
