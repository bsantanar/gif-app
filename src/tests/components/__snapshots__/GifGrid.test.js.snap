// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for <GifGrid /> should match snapshot 1`] = `
<Fragment>
  <h3>
    Jojo
  </h3>
  <div
    className="card-grid"
  />
</Fragment>
`;
